app/common/model/src/main/java/io/syndesis/common/model/WithMetadata.java

```diff
@@ -34,4 +34,10 @@
             getMetadata().get(key)
         );
     }
+
+    @JsonIgnore
+    default String getMetadata(String key, String defaultValue) {
+        return Optional.ofNullable(getMetadata().get(key))
+                       .orElse(defaultValue);
```

Comment:
I think this can be just `getMetadata().getOrDefault(key, defaultValue)`

Reply:
yes true, this helper method is pretty much not needed. I will rmeove